src/lib/registration/ui/addon_selection_base_dialog.rb

```diff
@@ -211,7 +219,8 @@ def handle_addon_selection(id)
         return unless addon
 
         show_addon_details(addon)
-        if Yast::UI.QueryWidget(Id(addon_widget_id(addon)), :Value)
+        new_state = !addon.selected?
+        if new_state
```

Comment:
still my comment regrding simplifying to `addon.selected? ? addon.unselected : addon.selected` is valid